### PR TITLE
sci-electronics/librepcb: bump version to 0.1.5

### PR DESCRIPTION
**Pull request by ktrace**
- Bump to 0.1.5
- unbundle quazip

Package-Manager: Portage-3.0.13, Repoman-3.0.2
Closes: https://github.com/gentoo/gentoo/pull/19384
Signed-off-by: Victor Kustov <ktrace@yandex.ru>

### CODE REVIEW

File: sci-electronics/librepcb/librepcb-0.1.5.ebuild [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# Copyright 2009-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit qmake-utils xdg-utils
```
Comment (telans) [inline sci-electronics/librepcb/librepcb-0.1.5.ebuild:6]:
```suggestion
inherit qmake-utils xdg
```

File: sci-electronics/librepcb/librepcb-0.1.5.ebuild [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+	xdg_desktop_database_update
+	xdg_icon_cache_update
+	xdg_mimeinfo_database_update
+}
+
+pkg_postrm() {
+	xdg_desktop_database_update
+	xdg_icon_cache_update
+	xdg_mimeinfo_database_update
+}
```
Comment (telans) [inline sci-electronics/librepcb/librepcb-0.1.5.ebuild:59]:
```suggestion
```

**Comment by ktrace** [comment]
fixed

File: sci-electronics/librepcb/librepcb-0.1.5.ebuild [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+	dev-qt/qtcore:5
+	dev-qt/qtconcurrent:5
+	dev-qt/qtgui:5
+	dev-qt/qtnetwork:5[ssl]
+	dev-qt/qtopengl:5
+	dev-qt/qtprintsupport:5
+	dev-qt/qtsql:5
+	dev-qt/qtwidgets:5
+	dev-qt/qtxml:5
+	sys-libs/zlib"
```
Comment (juippis) [inline sci-electronics/librepcb/librepcb-0.1.5.ebuild:32]:
```
 * librepcb-0.1.5-source.zip BLAKE2B SHA512 size ;-) ...                                     [ ok ]
>>> Unpacking source...
>>> Unpacking librepcb-0.1.5-source.zip to /var/tmp/portage/sci-electronics/librepcb-0.1.5/work
>>> Source unpacked in /var/tmp/portage/sci-electronics/librepcb-0.1.5/work
>>> Preparing source in /var/tmp/portage/sci-electronics/librepcb-0.1.5/work/librepcb-0.1.5 ...
>>> Source prepared.
>>> Configuring source in /var/tmp/portage/sci-electronics/librepcb-0.1.5/work/librepcb-0.1.5 ...
 * Running qmake ...
Info: creating stash file /var/tmp/portage/sci-electronics/librepcb-0.1.5/work/librepcb-0.1.5/.qmake.stash
Reading /var/tmp/portage/sci-electronics/librepcb-0.1.5/work/librepcb-0.1.5/apps/apps.pro
 Reading /var/tmp/portage/sci-electronics/librepcb-0.1.5/work/librepcb-0.1.5/apps/librepcb/librepcb.pro
Project ERROR: Unknown module(s) in QT: svg                                                  [ !! ]
```
So some qt dep needs svg, I'm not sure which one.

We should keep an eye on #19565 and perhaps merge this after quazip change?

File: sci-electronics/librepcb/librepcb-0.1.5.ebuild [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+DEPEND="${RDEPEND}"
+
+src_configure() {
+	local pro_lst
+	for pro_lst in $(find libs/librepcb/ -type f -name "*.pro" )
```
Comment (juippis) [inline sci-electronics/librepcb/librepcb-0.1.5.ebuild:38]:
```suggestion
	for pro_lst in $(find libs/librepcb/ -type f -name "*.pro" || die)
```

File: sci-electronics/librepcb/librepcb-0.1.5.ebuild (hunk 3 above)
Comment (a17r) [reply sci-electronics/librepcb/librepcb-0.1.5.ebuild:32]: ebuild needs to add `dev-qt/qtsvg:5` unless it is a bogus dependency.

Comment (juippis) [reply sci-electronics/librepcb/librepcb-0.1.5.ebuild:32]: Well now I'm getting 
```
Project ERROR: quazip development package not found                                          [ !! ]

 * Running qmake has failed! (see above for details)
 * This shouldn't happen - please send a bug report to https://bugs.gentoo.org/

 * ERROR: sci-electronics/librepcb-0.1.5::testworld failed (configure phase):
 *   eqmake5 failed
```
after 257b4edee92e6a2aa211ec78251cc317bb6c3323

Could you check what are the workarounds for many packages in https://github.com/gentoo/gentoo/pull/19565/files

Also I don't see qtsvg added here, it seemed to be necessary on my last test run?

Comment (a17r) [reply sci-electronics/librepcb/librepcb-0.1.5.ebuild:32]: yes indeed, this has to be fixed. package uses PKGCONFIG to detect system quazip, so look no further than this for inspiration:

https://github.com/gentoo/gentoo/pull/19565/files#diff-def719ef07c52ab037e92ef945c897ce56ea6b467750839f0b7a3395b80a378c
https://github.com/gentoo/gentoo/pull/19565/files#diff-40f346d5b7768a23c642227cbb7ab6c01f068b9552870ca839ab014ff461619cR53

(this is first patching the upstream build system to use quazip1-qt5, then using sed to undo that change in case stable quazip is installed)

**Review by a17r — CHANGES_REQUESTED** [review]

**Review by juippis — COMMENTED** [review]
I still can't compile this. First the missing qtsvg, then the missing/broken quazip.

**Comment by a17r** [comment]
Also `dev-libs/quazip-0.9.1` is about to be removed so you could throw away any backwards compatibility and just fix buildsys/depend on >=1.1-r1.

**Comment by a17r** [comment]
ping / see also https://bugs.gentoo.org/741358

File: sci-electronics/librepcb/librepcb-0.1.5.ebuild [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+	dev-qt/qttest"
+
+PATCHES=( "${FILESDIR}"/${P}-quazip1.x-common-filedownload.patch )
+
+src_configure() {
+	local pro_lst
+	for pro_lst in $(find libs/librepcb/ -type f -name "*.pro" || die)
+	do
+		sed -e 's:target.path = $${PREFIX}/lib:target.path = $${PREFIX}/'$(get_libdir)':' \
+			-e '/PKGCONFIG/s:quazip:quazip1-qt5:' -i ${pro_lst} || die "Can't fix ${pro_lst} file"
```
Comment (a17r) [inline sci-electronics/librepcb/librepcb-0.1.5.ebuild:45]: why is this not part of the patch?

File: sci-electronics/librepcb/librepcb-0.1.5.ebuild [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+	dev-qt/qtopengl:5
+	dev-qt/qtprintsupport:5
+	dev-qt/qtsql:5
+	dev-qt/qtsvg:5
+	dev-qt/qtwidgets:5
+	dev-qt/qtxml:5
+	sys-libs/zlib"
+
+DEPEND="${RDEPEND}
+	dev-qt/qttest"
```
Comment (a17r) [inline sci-electronics/librepcb/librepcb-0.1.5.ebuild:36]: slot def here as well please.

**Review by a17r — CHANGES_REQUESTED** [review]
builds fine now, but still remaining nitpicks.

File: sci-electronics/librepcb/librepcb-0.1.5.ebuild (hunk 6 above)
Comment (ktrace) [reply sci-electronics/librepcb/librepcb-0.1.5.ebuild]: fixed

File: sci-electronics/librepcb/librepcb-0.1.5.ebuild (hunk 5 above)
Comment (ktrace) [reply sci-electronics/librepcb/librepcb-0.1.5.ebuild]: remake patch + sed
